syntax = "proto3";

package task;

message Task {
  int32 id = 1;
  string title = 2;
  string description = 3;
  int32 author_id = 4;
  bool completed = 5;
}

message CreateTaskRequest {
  string title = 1;
  string description = 2;
  int32 author_id = 3;
}

message UpdateTaskRequest {
  int32 id = 1;
  string title = 2;
  string description = 3;
  int32 author_id = 4;
  bool completed = 5;
}

message DeleteTaskRequest {
  int32 id = 1;
  int32 author_id = 2;
}

message GetTaskRequest {
  int32 id = 1;
}

message ListTasksRequest {
  int32 page = 1;
  int32 per_page = 2;
}

message TaskResponse {
  int32 status = 1;
  Task task = 2;
}

message ListTasksResponse {
  int32 status = 1;
  repeated Task tasks = 2;
  int32 total_count = 3;
}

service TaskService {
  rpc CreateTask(CreateTaskRequest) returns (TaskResponse);
  rpc UpdateTask(UpdateTaskRequest) returns (TaskResponse);
  rpc DeleteTask(DeleteTaskRequest) returns (TaskResponse);
  rpc GetTask(GetTaskRequest) returns (TaskResponse);
  rpc ListTasks(ListTasksRequest) returns (ListTasksResponse);
}
